lib/OpenQA/Utils.pm

```diff
@@ -310,7 +311,26 @@ sub bugurl {
 sub bugref_to_href {
     my ($text) = @_;
 
-    $text =~ s{((bnc|bsc|boo|poo)#(\d+))}{<a href="@{[bugurl($2)]}$3">$1</a>}gi;
+    $text =~ s{((bnc|bsc|boo|poo)#(\d+)( |$))}{<a href="@{[bugurl($2)]}$3">$1</a>}gim;
+
+    return $text;
+}
+
+sub bugabbr {
+    my ($bugurl) = @_;
+    my %bugurls = (
+        'https://bugzilla.novell.com/show_bug.cgi?id='   => 'bnc',
+        'https://bugzilla.suse.com/show_bug.cgi?id='     => 'bsc',
+        'https://bugzilla.opensuse.org/show_bug.cgi?id=' => 'boo',
+        'https://progress.opensuse.org/issues/'          => 'poo',
+    );
+    return $bugurls{$bugurl};
+}
+
+sub href_to_bugref {
+    my ($text) = @_;
+
+    $text =~ s{((https://bugzilla.novell.com/show_bug.cgi?id=|https://bugzilla.suse.com/show_bug.cgi?id=|https://bugzilla.opensuse.org/show_bug.cgi?id=|https://progress.opensuse.org/issues/)(\d+)( |$))}{@{[bugabbr($2)]}#$3}gim;
```

Comment:
I'm sorry, but no.
1) the regex is wrong, you need to escape `?`, try to add some different url than poo to the tests.
2) this is 3rd time copying basically the same info. You can build reverse lookup hash dynamically from the original %bugref hash. The same for regexp.

Something like:
```perl
sub _bugref_lookup {
    my ($url) = @_;
    my @res = grep { $bugrefs{$_} eq $url } keys %bugrefs;
    return $res[0];
}

sub href_to_bugref {
    my ($text) = @_;

    my @urlregexes;
    for (values %bugrefs) {
       # escape ?
       my ($r) = $_ =~ s/\?/\\?/gr;
       push @urlregexes, $r;
    }

    my $regex = join('|', @urlregexes);
    $regex = qr/($regex)(\d+)( |$)/;
    $text =~ s{$regex}{@{[_bugref_lookup($1)]}#$2$3}gim;
    return $text;
}
```